icu4j/main/classes/core/src/com/ibm/icu/impl/units/SingleUnitImpl.java

```diff
@@ -122,21 +122,49 @@ int compareTo(SingleUnitImpl other) {
         if (index > other.index) {
             return 1;
         }
-        // TODO: revisit if the spec dictates prefix sort order - it doesn't
-        // currently. For now we're sorting binary prefixes before SI prefixes,
-        // as per ICU4C's enum values order.
-        if (this.getPrefix().getBase() < other.getPrefix().getBase()) {
+
+        // When comparing binary prefixes vs SI prefixes, instead of comparing the actual values, we can
+        // multiply the binary prefix power by 3 and compare the powers. if they are equal, we can can
+        // compare the bases.
+        // NOTE: this methodology will fail if the binary prefix more than or equal 98.
+        int unitBase = this.unitPrefix.getBase();
+        int otherUnitBase = other.unitPrefix.getBase();
+        // Values for comparison purposes only.
+        int unitPowerCompAbs =
+                unitBase == 1024 /* Binary Prefix */ ? Math.abs(this.unitPrefix.getPower()) * 3
+                        : Math.abs(this.unitPrefix.getPower());
```

Comment:
As with C++, I don't think taking absolute values makes sense?

Reply:
I had forgotten I had even reviewed some Java code. :blush: But I've already spent more time on this than I can afford before perf. Can I pass this on to @sffc to review the Java code / ensuring it matches C++?